ppx/test/constant-detection.t/input.re

```diff
@@ -0,0 +1,14 @@
+module Foo = {
+  [@react.component]
+  let make = (~id) => {
+    <div key=id> [|<div />, "a"|] "a" 'b' 3 false </div>;
```

Comment:
What happens in the following cases?
- Child is a variant or polyvar with a string payload
- Child is a quoted string
- Child is a tuple of strings
- Child is a function application with a string param
- Child is an extension with an string payload, e.g. `[%intl.s "foo"]`, we use these a lot at Ahrefs

Reply:
This only checks the Pconst kinds and applies the right React.string/int/float/char.

For all your cases, the type-checker will break. I would like to see the quality of the error in such cases

Reply:
yeah, if we move ahead with this, proper testing is required. I just included a simple cram syntactical example to show case what the PPX desugars to.